https/tls_config.go

```diff
@@ -168,3 +224,44 @@ func Listen(server *http.Server, tlsConfigPath string, logger log.Logger) error
 	}
 	return server.ListenAndServeTLS("", "")
 }
+
+// For go1.14, unmarshalling is done properly in tls_ciphers.go.
+type cipher uint16
+
+type curve tls.CurveID
+
+func (c *curve) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var s string
+	err := unmarshal((*string)(&s))
+	if err != nil {
+		return err
+	}
+	switch strings.ToUpper(s) {
+	case "CURVEP256":
+		*c = (curve)(tls.CurveP256)
+	case "CURVEP384":
+		*c = (curve)(tls.CurveP384)
+	case "CURVEP521":
+		*c = (curve)(tls.CurveP521)
+	case "X25519":
+		*c = (curve)(tls.X25519)
+	default:
+		return errors.New("unknown supported group: " + s)
+	}
+	return nil
+}
+
+func (c curve) MarshalYAML() (interface{}, error) {
+	switch c {
+	case (curve)(tls.CurveP256):
```

Comment:
Might a map be better rather than repeating the curves twice?

Reply:
Done